internal/cli/atlas/backup/compliancepolicy/setup.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compliancepolicy
+
+import (
+	"context"
+	"errors"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type SetupOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	policy  *atlasv2.DataProtectionSettings
+	store   store.CompliancePolicy
+	fs      afero.Fs
+	path    string
+	confirm bool
+}
+
+var setupTemplate = `
+Your backup compliance policy has been set up with the following configuration:
+
+Project:	{{.ProjectId}}
+Authorized e-mail:	{{.AuthorizedEmail}}
+Copy protection enabled:	{{.CopyProtectionEnabled}}
+Encryption at rest enabled:	{{.EncryptionAtRestEnabled}}
+Point-in-Time restores enabled:	{{.PitEnabled}}
+Restore window days:	{{.RestoreWindowDays}}
+
+POLICIES
+ID	FREQUENCY INTERVAL	FREQUENCY TYPE	RETENTION
+{{- range .ScheduledPolicyItems}}
+{{.Id}}	{{if eq .FrequencyType "hourly"}}{{.FrequencyInterval}}{{else}}-{{end}}	{{.FrequencyType}}	{{.RetentionValue}} {{.RetentionUnit}}
+{{- end}}
+{{if .OnDemandPolicyItem}}{{.OnDemandPolicyItem.Id}}	-	{{.OnDemandPolicyItem.FrequencyType}}	{{.OnDemandPolicyItem.RetentionValue}} {{.OnDemandPolicyItem.RetentionUnit}}{{end}}
+`
+
+func (opts *SetupOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *SetupOpts) setupWatcher() (bool, error) {
+	res, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	opts.policy = res
+	if err != nil {
+		return false, err
+	}
+	if res.GetState() == "" {
+		return false, errors.New("could not access State field")
+	}
+	return (res.GetState() == active), nil
+}
+
+func (opts *SetupOpts) Run() error {
+	if !opts.confirm {
+		return errors.New("this feature is not production ready: to continue, use '--force'")
+	}
```

Comment:
Right now we throw this error until the interactive confirmation flow is implemented. I am open to a better error, but too much effort shouldn't be wasted – it'll be removed in no time.

Reply:
don't add the flag would be a better way to handle this ;)

Reply:
Agreed. Will remove.